### PR TITLE
Calculate normalized freqs in value_counts correctly when bins is not None

### LINKED ISSUE #25970: value_counts unexpected behaviour - bins and dropna
#### 1st example:

Input:
```python
series = pd.Series([1, 1, 2, 0, 1, np.nan, 4, 4, np.nan, 3])
series.value_counts(True, bins=2)

```

Output:
```python
(-0.005, 2.0]    0.5
(2.0, 4.0]       0.3
dtype: float64
```

Expected output:
```python
(-0.005, 2.0]    0.625
(2.0, 4.0]       0.375
dtype: float64

```

#### 2nd example:

Input:
```python
series = pd.Series([1, 1, 2, 0, 1, np.nan, 4, 4, np.nan, 3])
series.value_counts(True, bins=2, dropna=False)

```

Output:
```python
(-0.005, 2.0]    0.5
(2.0, 4.0]       0.3
dtype: float64
```

Expected Output:
```python
(-0.004, 2.0]    0.5
(2.0, 4.0]       0.3
NaN              0.2
dtype: float64

```

#### Problem description
dropna argument in value_counts() seems to have no effect when bins is not None. Expected behaviour is better, because it sums up shares up to 1 when dropna=True and shows NaNs when dropna=False.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.24.2
pytest: None
pip: 19.0.3
setuptools: 40.8.0
Cython: None
numpy: 1.16.2
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

### PR DESCRIPTION
**Pull request by krsnik93**
- [x] closes #25970
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_algos.py [hunk 1]
```diff
@@ -967,21 +967,28 @@ def test_dropna(self):
...
             tm.assert_series_equal(result, expected)
 
-    def test_value_counts_normalized(self):
+    @pytest.mark.parametrize('dropna, vals, index', [
+        (False, [0.6, 0.2, 0.2], [np.nan, 2.0, 1.0]),
+        (True, [0.5, 0.5], [2.0, 1.0])])
+    def test_value_counts_normalized(self, dropna, vals, index):
         # GH12558
         s = Series([1, 2, np.nan, np.nan, np.nan])
         dtypes = (np.float64, np.object, 'M8[ns]')
```
Comment (jreback) [inline pandas/tests/test_algos.py]: can you pull the dtypes into another level of parameterization

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew note, reshape section of bug fixes

**Comment by jreback** [comment]
@krsnik93 can you merge master & add a whatsnew note (bug fix in reshaping is ok). ping on green.

**Comment by krsnik93** [comment]
I have a failing test in tests/groupby/test_value_counts.py. The test compares `Series.value_counts` against `SeriesGroupBy.value_counts`. For this issue, I modified Series.value_counts and I think the change is correct. However, I think `SeriesGroupBy.value_counts` should change as well. This is the troubling bit:

https://github.com/pandas-dev/pandas/blob/c275dbfcee0fa9644fa2718ad76fd91ca056069b/pandas/core/groupby/generic.py#L1261-L1275

For some reason, it takes by-group counts of unbinned data (either NaN values or outside of bin breaks) and resets those counts to 0 (last line). I do not understand why is this done, but the whole method is kind hard to read and I am not able to make the required changes.

**Comment by TomAugspurger** [comment]
cc @jschendel if you have time to take a look.

**Comment by WillAyd** [comment]
@krsnik93 can you merge master and fix up merge conflicts?

**Comment by WillAyd** [comment]
Closing as this looks stale, but @krsnik93 please ping if you'd like to continue and can open back up

**Pull request closed without merging**